### PR TITLE
chore(edit-content): For Image Field only image files should be displayed.

### LINKED ISSUE #30060: Image Field: "Select Existing File" functionality
### Parent Issue

https://github.com/dotCMS/core/issues/29692

### User Story

As a content editor, I want to select an existing file and have the system display only image files, so that I can quickly find and choose the appropriate image for my content.

### Acceptance Criteria

1. In the File field, there is a "Select Existing File" option.
2. Clicking on this option opens a popup
3. Hosts/Folders should appear on the left side.
4. Navigate through the levels
5. Select a folder/host
6. Only Image Files should be loaded

### PR DESCRIPTION
**Pull request by nicobytes**
### Parent Isssue

https://github.com/dotCMS/core/issues/30060

### Proposed Changes

This pull request includes several updates to the `core-web/libs/edit-content` library, focusing on refactoring and improving the file field preview and selection components. The most important changes include the introduction of a new `FileInfo` type, updates to the `DotFileFieldPreviewComponent`, and modifications to the `DotSelectExistingFileComponent` and its store.

### Refactoring and Type Definitions:

* [`core-web/libs/edit-content/src/lib/fields/dot-edit-content-file-field/components/dot-file-field-preview/dot-file-field-preview.component.ts`](diffhunk://#diff-9d3250303440cce67e959a59ed17f44ad49d690c04e36cbf4fadc011f206001eR37-R46): Introduced `FileInfo` type to consolidate file-related properties and updated the component to use this new type. [[1]](diffhunk://#diff-9d3250303440cce67e959a59ed17f44ad49d690c04e36cbf4fadc011f206001eR37-R46) [[2]](diffhunk://#diff-9d3250303440cce67e959a59ed17f44ad49d690c04e36cbf4fadc011f206001eL73-R114)
* [`core-web/libs/edit-content/src/lib/fields/dot-edit-content-file-field/components/dot-form-import-url/dot-form-import-url.component.ts`](diffhunk://#diff-5efc519c0e3b6fc75fe06f8115527c72874b15e764a0c05508e341d0643ca624L15-R24): Added `DialogData` type for better type safety in dialog configurations. [[1]](diffhunk://#diff-5efc519c0e3b6fc75fe06f8115527c72874b15e764a0c05508e341d0643ca624L15-R24) [[2]](diffhunk://#diff-5efc519c0e3b6fc75fe06f8115527c72874b15e764a0c05508e341d0643ca624L39-R44)

### Component Updates:

* [`core-web/libs/edit-content/src/lib/fields/dot-edit-content-file-field/components/dot-file-field-preview/dot-file-field-preview.component.html`](diffhunk://#diff-64ed272338ca8810a9254c03bad8f295b77e25c76d92cf3db5c76038ed215474L1-R20): Refactored to use `fileInfo` instead of multiple individual variables for file properties. [[1]](diffhunk://#diff-64ed272338ca8810a9254c03bad8f295b77e25c76d92cf3db5c76038ed215474L1-R20) [[2]](diffhunk://#diff-64ed272338ca8810a9254c03bad8f295b77e25c76d92cf3db5c76038ed215474L59-R63) [[3]](diffhunk://#diff-64ed272338ca8810a9254c03bad8f295b77e25c76d92cf3db5c76038ed215474L82-R86)
* [`core-web/libs/edit-content/src/lib/fields/dot-edit-content-file-field/components/dot-file-field-preview/dot-file-field-preview.component.ts`](diffhunk://#diff-9d3250303440cce67e959a59ed17f44ad49d690c04e36cbf4fadc011f206001eL129-R135): Updated the component initialization and methods to work with the new `FileInfo` type. [[1]](diffhunk://#diff-9d3250303440cce67e959a59ed17f44ad49d690c04e36cbf4fadc011f206001eL129-R135) [[2]](diffhunk://#diff-9d3250303440cce67e959a59ed17f44ad49d690c04e36cbf4fadc011f206001eL163-R169)

### Store and Service Updates:

* [`core-web/libs/edit-content/src/lib/fields/dot-edit-content-file-field/components/dot-select-existing-file/store/select-existing-file.store.ts`](diffhunk://#diff-4e8a1b332dae5fb8ead291b76bf6099437aaf4aa6fd0a759ffe71c3074ffe7f0L2-R9): Added `mimeTypes` state and related methods to handle file type filtering. [[1]](diffhunk://#diff-4e8a1b332dae5fb8ead291b76bf6099437aaf4aa6fd0a759ffe71c3074ffe7f0L2-R9) [[2]](diffhunk://#diff-4e8a1b332dae5fb8ead291b76bf6099437aaf4aa6fd0a759ffe71c3074ffe7f0L40-R50) [[3]](diffhunk://#diff-4e8a1b332dae5fb8ead291b76bf6099437aaf4aa6fd0a759ffe71c3074ffe7f0L57-R67) [[4]](diffhunk://#diff-4e8a1b332dae5fb8ead291b76bf6099437aaf4aa6fd0a759ffe71c3074ffe7f0R80-R84)
* [`core-web/libs/edit-content/src/lib/fields/dot-edit-content-file-field/components/dot-select-existing-file/store/select-existing-file.store.test.ts`](diffhunk://#diff-547a685cc26753b615660d1f7014944ee45e352b8629ba7daa2d10522fa6c77dL16-R32): Updated tests to use `editContentService` instead of `service` for better clarity and consistency. [[1]](diffhunk://#diff-547a685cc26753b615660d1f7014944ee45e352b8629ba7daa2d10522fa6c77dL16-R32) [[2]](diffhunk://#diff-547a685cc26753b615660d1f7014944ee45e352b8629ba7daa2d10522fa6c77dL33-R41) [[3]](diffhunk://#diff-547a685cc26753b615660d1f7014944ee45e352b8629ba7daa2d10522fa6c77dL44-R52) [[4]](diffhunk://#diff-547a685cc26753b615660d1f7014944ee45e352b8629ba7daa2d10522fa6c77dL59-R67) [[5]](diffhunk://#diff-547a685cc26753b615660d1f7014944ee45e352b8629ba7daa2d10522fa6c77dL79-R87)

### Configuration Changes:

* [`core-web/libs/edit-content/jest.config.ts`](diffhunk://#diff-a5a47422d33499f0f3d15a48bc9a17e7b3af763708f81899272096fd5457550dL21-R22): Added `@happy-dom/jest-environment` as the test environment for better compatibility with DOM-related tests.

### Checklist
- [x] Tests
- [x] Translations
- [x] Security Implications Contemplated (add notes if applicable)

### Result

1. Image Field: When the user clicks on the “Select Existing File” option, only image files should be displayed.
2. File Field: When the user clicks on the “Select Existing File” option, all types of content should be displayed.

https://github.com/user-attachments/assets/83c6ce24-015c-4577-af10-5354eb32a11e

This PR fixes: #30060

### CODE REVIEW

File: core-web/libs/edit-content/src/lib/fields/dot-edit-content-file-field/components/dot-select-existing-file/store/select-existing-file.store.ts [hunk 1]
```diff
@@ -186,5 +203,11 @@ export const SelectExisingFileStore = signalStore(
                 )
             )
         };
-    })
+    }),
+    withHooks((store) => ({
```
Comment (KevinDavilaDotCMS) [inline core-web/libs/edit-content/src/lib/fields/dot-edit-content-file-field/components/dot-select-existing-file/store/select-existing-file.store.ts:207]: TIL

File: core-web/libs/edit-content/jest.config.ts [hunk 2]
```diff
@@ -18,5 +18,6 @@ export default {
         'jest-preset-angular/build/serializers/no-ng-attributes',
         'jest-preset-angular/build/serializers/ng-snapshot',
         'jest-preset-angular/build/serializers/html-comment'
-    ]
+    ],
+    testEnvironment: '@happy-dom/jest-environment'
```
Comment (zJaaal) [inline core-web/libs/edit-content/jest.config.ts:22]: What does this does?

Comment (nicobytes) [reply core-web/libs/edit-content/jest.config.ts:22]: There is an [issue](https://github.com/primefaces/primeng/issues/14085) with Jest and PrimeNG when using the `@layer` CSS feature. The root cause of this issue is that JSDOM does not support this property. A possible solution is to use the `@happy-dom/jest-environment`, which provides support for the `@layer` CSS feature.

**Review by hmoreras — APPROVED** [review]

File: core-web/libs/edit-content/src/lib/services/dot-edit-content.service.ts [hunk 3]
```diff
@@ -227,17 +227,18 @@ export class DotEditContentService {
             .pipe(map((response) => response.entity.count));
     }
```
[the remark this answers is not in the document]
Comment (oidacra) [reply core-web/libs/edit-content/src/lib/services/dot-edit-content.service.ts:229]: And a test.

File: core-web/libs/edit-content/src/lib/models/dot-edit-content-file.model.ts [hunk 4]
```diff
@@ -1,6 +1,12 @@
 import { DotCMSContentlet, DotCMSTempFile } from '@dotcms/dotcms-models';
 
-export type INPUT_TYPES = 'File' | 'Image' | 'Binary';
+export enum INPUT_TYPES {
+    File = 'File',
+    Image = 'Image',
+    Binary = 'Binary'
+}
+
+export type INPUT_TYPE = `${INPUT_TYPES}`;
```
Comment (oidacra) [inline core-web/libs/edit-content/src/lib/models/dot-edit-content-file.model.ts:9]: Cool.

File: core-web/libs/edit-content/src/lib/fields/dot-edit-content-file-field/dot-edit-content-file-field.component.ts [hunk 5]
```diff
@@ -406,7 +406,10 @@ export class DotEditContentFileFieldComponent implements ControlValueAccessor, O
             modal: true,
             width: '90%',
             style: { 'max-width': '1040px' },
-            data: {}
+            data: {
```
Comment (oidacra) [inline core-web/libs/edit-content/src/lib/fields/dot-edit-content-file-field/dot-edit-content-file-field.component.ts:409]: Add doc to this public method

File: core-web/libs/edit-content/src/lib/services/dot-edit-content.service.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (nicobytes) [reply core-web/libs/edit-content/src/lib/services/dot-edit-content.service.ts:229]: Fixed!

File: core-web/libs/edit-content/src/lib/fields/dot-edit-content-file-field/dot-edit-content-file-field.component.ts (hunk 5 above)
Comment (nicobytes) [reply core-web/libs/edit-content/src/lib/fields/dot-edit-content-file-field/dot-edit-content-file-field.component.ts:417]: Fixed!

**Review by oidacra — APPROVED** [review]

**Pull request merged**